presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSink.java

```diff
@@ -294,12 +355,46 @@ private int[] getWriterIndexes(Page page)
         Page transformedPage = pagePartitioner.getTransformedPage();
         for (int position = 0; position < page.getPositionCount(); position++) {
             int writerIndex = writerIndexes[position];
-            if (writers.get(writerIndex) != null) {
+            WriteContext writer = writers.get(writerIndex);
+            if (writer != null) {
                 continue;
             }
 
-            Optional<PartitionData> partitionData = getPartitionData(pagePartitioner.getColumns(), transformedPage, position);
-            WriteContext writer = createWriter(partitionData);
+            Optional<PartitionData> partitionData = getPartitionData(pagePartitioner.getColumns(), page, position);
```

Comment:
```suggestion
            Optional<PartitionData> partitionData = getPartitionData(pagePartitioner.getColumns(), transformedPage, position);
```

After check the code, I think the problem is introduced by this change.

Reply:
Thanks @hantangwangd . that was the issue. done the changes. Will let you know once PR is ready for review.